libraries/extractor/src/main/java/androidx/media3/extractor/mp4/BoxParser.java

```diff
@@ -860,6 +866,36 @@ private static Metadata parseXyz(ParsableByteArray xyzBox) {
     }
   }
 
+  /** Parses the chapter metadata from the chpl atom. */
+  @Nullable
+  /* package */ static Metadata parseChpl(ParsableByteArray chpl) {
+    chpl.skipBytes(Mp4Box.HEADER_SIZE);
+    chpl.skipBytes(5); // version (1) + flags (3) + reservered byte (1)
+    ArrayList<Metadata.Entry> entries = new ArrayList<>();
+    int chapterCount = chpl.readInt();
+    for (int i = 0; i < chapterCount; i++) {
+      long startTimeMs = chpl.readLong() / 10_000; // Start time in 100-nanoseconds resolution
+      if (startTimeMs < 0 || startTimeMs > Integer.MAX_VALUE) {
+        startTimeMs = C.INDEX_UNSET;
+      }
+      int titleLength = chpl.readUnsignedByte();
+      String chapterName = chpl.readString(titleLength);
+      ChapterFrame chapterFrame =
+          new ChapterFrame(
+              /* chapterId= */ Integer.toString(i),
```

Comment:
I wasn't sure if something else should be used for an ID. MP4 chapters do not have this concept.